stan/math/rev/functor/idas_forward_system.hpp

```diff
@@ -121,56 +121,48 @@ class idas_forward_system : public idas_system<F, Tyy, Typ, Tpar> {
       auto yys_mat = matrix_d_from_NVarray(yys, ns);
       auto yps_mat = matrix_d_from_NVarray(yps, ns);
 
-      try {
-        stan::math::start_nested();
-
-        MatrixXd J, r;
-        VectorXd f_val;
-
-        auto fyy
-            = [&t, &vyp, &vtheta, &N, &dae](const matrix_v& x) -> vector_v {
-          std::vector<var> yy(x.data(), x.data() + N);
-          auto eval
-              = dae->f_(t, yy, vyp, vtheta, dae->x_r_, dae->x_i_, dae->msgs_);
-          Eigen::Map<vector_v> res(eval.data(), N);
-          return res;
-        };
-        stan::math::jacobian(fyy, vec_yy, f_val, J);
-        r = J * yys_mat;
-
-        auto fyp
-            = [&t, &vyy, &vtheta, &N, &dae](const matrix_v& x) -> vector_v {
-          std::vector<var> yp(x.data(), x.data() + N);
+      // Run nested autodiff in this scope
+      stan::math::nested_rev_autodiff nested;
+
+      MatrixXd J, r;
+      VectorXd f_val;
+
+      auto fyy = [&t, &vyp, &vtheta, &N, &dae](const matrix_v& x) -> vector_v {
+        std::vector<var> yy(x.data(), x.data() + N);
+        auto eval
+            = dae->f_(t, yy, vyp, vtheta, dae->x_r_, dae->x_i_, dae->msgs_);
+        Eigen::Map<vector_v> res(eval.data(), N);
+        return res;
+      };
+      stan::math::jacobian(fyy, vec_yy, f_val, J);
+      r = J * yys_mat;
+
+      auto fyp = [&t, &vyy, &vtheta, &N, &dae](const matrix_v& x) -> vector_v {
+        std::vector<var> yp(x.data(), x.data() + N);
+        auto eval
+            = dae->f_(t, vyy, yp, vtheta, dae->x_r_, dae->x_i_, dae->msgs_);
+        Eigen::Map<vector_v> res(eval.data(), N);
+        return res;
+      };
+      stan::math::jacobian(fyp, vec_yp, f_val, J);
+      r += J * yps_mat;
+
+      if (dae->is_var_par) {
+        auto fpar
+            = [&t, &vyy, &vyp, &N, &M, &dae](const matrix_v& x) -> vector_v {
+          std::vector<var> par(x.data(), x.data() + M);
           auto eval
-              = dae->f_(t, vyy, yp, vtheta, dae->x_r_, dae->x_i_, dae->msgs_);
+              = dae->f_(t, vyy, vyp, par, dae->x_r_, dae->x_i_, dae->msgs_);
```

Comment:
[question]
Why did this call change when this PR is just about nested autodiff?

Reply:
It didn't change - note that the diff is misaligned and line 145 on the left should actually be paired with line 143 on the right. Possibly the reformatting by clangformat misled the diff algorithm. If you view the diff ignoring whitespace (can be set via the cog icon at the top), it is visible that the code didn't change.